src/storage/query/QueryEdgeKeysProcessor.cpp

```diff
@@ -7,48 +7,52 @@
 #include "base/NebulaKeyUtils.h"
 #include "storage/query/QueryEdgeKeysProcessor.h"
 
-
 namespace nebula {
 namespace storage {
 
-void QueryEdgeKeysProcessor::process(const cpp2::EdgeKeyRequest& req) {
-    auto spaceId = req.get_space_id();
-    auto vId = req.get_vid();
-    auto partId = req.get_part_id();
+void QueryEdgeKeysProcessor::process(const cpp2::EdgeKeysRequest& req) {
     CHECK_NOTNULL(kvstore_);
+    auto spaceId = req.get_space_id();
+    const auto& partVertices = req.get_parts();
+
+    std::unordered_map<VertexID, std::vector<cpp2::EdgeKey>> edge_keys;
+    std::for_each(partVertices.begin(), partVertices.end(), [&](auto& pv) {
+        auto part = pv.first;
+        const auto& vertices = pv.second;
+        std::for_each(vertices.begin(), vertices.end(), [&](auto& v) {
+            auto prefix = NebulaKeyUtils::edgePrefix(part, v);
+            std::unique_ptr<kvstore::KVIterator> iter;
+            auto ret = this->kvstore_->prefix(spaceId, part, prefix, &iter);
+            if (ret != kvstore::ResultCode::SUCCEEDED) {
+                VLOG(3) << "Error! ret = " << static_cast<int32_t>(ret) << ", spaceId = " << spaceId
+                        << ", partId =  " << part << ", vertexId = " << v;
+                this->pushResultCode(this->to(ret), part);
+                this->onFinished();
+                return;
```

Comment:
Here you want to directly return of function `process`? but the return in `for_each` can't .